src/ip/Address.cc

```diff
@@ -404,7 +404,12 @@ Ip::Address::lookupHostIP(const char *s, bool nodns)
         }
         if (maybeIpv4 != NULL)
             res = maybeIpv4;
-        // else IPv6-only host, let the caller deal with first-IP anyway.
+
+        // leave this object unchanged if an IPv6 address was found
+        if (res->ai_family == AF_INET6) {
+            freeaddrinfo(resHead);
+            return false;
```

Comment:
This is a partial review -- I have not revisited earlier change requests and have not fixed the PR title yet.

The two `if` statements in the current PR code imply that there are three kinds of address families that the caller may be able to handle:

* IPv4: always returned to the caller.
* IPv6: returned to the caller only when Ip::EnableIpv6 is true.
* other: always returned to the caller.

I think it would be better to preserve the logic implied by the (removed) "else IPv6-only" comment in the official code because a lot of Squid code assumes that there are only two kinds of IP addresses:

* IPv4: always returned to the caller.
* other: returned to the caller only when Ip::EnableIpv6 is true.

That can be easily accomplished by replacing the new `if` statement with `else` AFAICT.

IMO, it would be better to tighten this up further, but that may be considered as going beyond this PR scope:

* IPv4: always returned to the caller.
* IPv6: returned to the caller only when Ip::EnableIpv6 is true.
* other: never returned to the caller.

Reply:
"other" / future address formats require IPv6 address handling. see RFC6874 and RFC3986 IPvFuture syntax.

Reply:
> "other" / future address formats require IPv6 address handling. see RFC6874 and RFC3986 IPvFuture syntax.

I cannot reach the same "require IPv6 address handling" conclusion after reading about IPvFuture in those two RFCs, but the conclusion depends on what "handling" is in this context, and I do not know how you define this term.

More importantly, I believe both of my change request suggestions (and their rationale) are compatible with _either_ conclusion, so we do not need to reach an agreement regarding those RFC implications. And I still think the second suggestion is better than the first one (but may be considered out of scope).

When it is time to support IPvFuture, whatever that ends up being, we will adjust this code as needed again (and decide whether true EnableIpv6 setting is a prerequisite for that next IP version support), of course. Hopefully, EnableIpv6 will be simply gone by then :-).